File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 0
    hash_map={}
    for cloth in clothes:
        # print(cloth[0], cloth[1])
        hash_map[cloth[1]] = []
    for cloth in clothes:
        hash_map[cloth[1]].append(cloth[0])
    print(hash_map)
    total = 1
    for kind in hash_map:
        total *= (len(hash_map[kind])+1)
    answer = total -1
    return answer

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map <string, vector<string>> mp;
    for(int i=0; i<clothes.size(); i++){
        string category = clothes[i][1];
        string cloth = clothes[i][0];
        mp[category].push_back(cloth);
    }
    for(auto a:mp){
        answer *= (a.second.size()+1);
    }
    return answer-1;
}